src/libraries/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Marshalling/OleVariantMarshaller.cs

```diff
@@ -0,0 +1,332 @@
+            {
+                variant = OleVariant.CreateRaw(VarEnum.VT_UNKNOWN, StrategyBasedComWrappers.DefaultMarshallingInstance.GetOrCreateComInterfaceForObject(managed, CreateComInterfaceFlags.None));
+                return true;
+            }
+            variant = default;
+            return false;
+        }
+#pragma warning restore CA1416 // Validate platform compatibility
+
+        public static unsafe object? ConvertToManaged(OleVariant unmanaged)
+        {
+#pragma warning disable CS0618 // Type or member is obsolete
+#pragma warning disable CA1416 // Validate platform compatibility
+            switch (unmanaged.VarType)
+            {
+                case VarEnum.VT_EMPTY:
+                case VarEnum.VT_BYREF | VarEnum.VT_EMPTY:
+                    return null;
+                case VarEnum.VT_NULL:
+                case VarEnum.VT_BYREF | VarEnum.VT_NULL:
+                    return DBNull.Value;
+                case VarEnum.VT_I1:
+                    return unmanaged.As<sbyte>();
+                case VarEnum.VT_UI1:
+                    return unmanaged.As<byte>();
+                case VarEnum.VT_I2:
+                    return unmanaged.As<short>();
+                case VarEnum.VT_UI2:
+                    return unmanaged.As<ushort>();
+                case VarEnum.VT_INT:
+                case VarEnum.VT_I4:
+                    return unmanaged.As<int>();
+                case VarEnum.VT_UINT:
+                case VarEnum.VT_UI4:
+                    return unmanaged.As<uint>();
+                case VarEnum.VT_I8:
+                    return unmanaged.As<long>();
+                case VarEnum.VT_UI8:
+                    return unmanaged.As<ulong>();
+                case VarEnum.VT_R4:
+                    return unmanaged.As<float>();
+                case VarEnum.VT_R8:
+                    return unmanaged.As<double>();
+                case VarEnum.VT_DECIMAL:
+                    return unmanaged.As<decimal>();
+                case VarEnum.VT_BOOL:
+                    return unmanaged.As<bool>();
+                case VarEnum.VT_BSTR:
+                    return unmanaged.As<string>();
+                case VarEnum.VT_DATE:
+                    return unmanaged.As<DateTime>();
+                case VarEnum.VT_ERROR:
+                    return unmanaged.As<int>();
+                case VarEnum.VT_CY:
+                    return unmanaged.As<CurrencyWrapper>().WrappedObject;
+                case VarEnum.VT_UNKNOWN:
+                case VarEnum.VT_DISPATCH:
+                    return StrategyBasedComWrappers.DefaultMarshallingInstance.GetOrCreateObjectForComInstance(unmanaged.GetRawDataRef<nint>(), CreateObjectFlags.Unwrap);
+                case VarEnum.VT_BYREF | VarEnum.VT_VARIANT:
+                    return ConvertToManaged(*(OleVariant*)unmanaged.GetRawDataRef<nint>());
+                case VarEnum.VT_BYREF | VarEnum.VT_I1:
+                    return *(sbyte*)unmanaged.GetRawDataRef<nint>();
```

Comment:
If `unmanaged` is a value type, that is passed into this function on the stack, how can we return a `ref` to one of its fields?